### PR TITLE
Add Ivy support

### PR DESCRIPTION
**Pull request by SchnWalter**
Adds support for Ivy and also updates requirements to Anguar and Angular Material v9.

### CODE REVIEW

File: saturn-datepicker/src/datepicker/tsconfig-build.json [hunk 1]
```diff
@@ -1,15 +0,0 @@
...
-  "extends": "../tsconfig-build",
-  "files": [
-    "public-api.ts",
-    "../typings.d.ts"
-  ],
-  "angularCompilerOptions": {
-    "annotateForClosureCompiler": true,
-    "strictMetadataEmit": true,
-    "flatModuleOutFile": "index.js",
-    "flatModuleId": "@angular/material/datepicker",
```
Comment (SchnWalter) [inline saturn-datepicker/src/datepicker/tsconfig-build.json]: NOTE: The library had a invalid flatModuleId, so I cleanned up all related code and also removed the invalid bazel build configuration.

File: package.json [hunk 2]
```diff
@@ -4,56 +4,59 @@
...
-    "ng-packagr": "^5.5.0",
-    "protractor": "^5.4.2",
-    "rimraf": "^2.7.1",
-    "ts-node": "~7.0.0",
-    "tsickle": "0.37.0",
-    "tslib": "^1.9.0",
-    "tslint": "~5.11.0",
-    "typescript": "3.5.3"
+    "@angular-devkit/build-angular": "0.900.4",
+    "@angular-devkit/build-ng-packagr": "0.900.4",
```
Comment (SchnWalter) [inline package.json]: Note: The library is now built using the Angular CLI, the workspace project configuration is named `lib` and the default project is set to `dev` (so `ng serve`, `ng build` will run against the development application).


To build the library for a release, just run:
```
ng build --project=lib --configuration=production
# or
ng run lib:build:production
```

File: saturn-datepicker/src/datepicker/README.md [hunk 3]
```diff
@@ -1 +0,0 @@
-Please see the official documentation at https://material.angular.io/components/component/datepicker
```
Comment (SaturnTeam) [inline saturn-datepicker/src/datepicker/README.md]: Why?

Comment (SaturnTeam) [reply saturn-datepicker/src/datepicker/README.md]: @SchnWalter

Comment (SchnWalter) [reply saturn-datepicker/src/datepicker/README.md]: Reverted the change.

**Comment by smurfy** [comment]
any update on this?

**Comment by absalonv** [comment]
any update on this?

**Comment by absalonv** [comment]
> any update on this?

I updated to  the latest version of **Angular** and latest of **saturn-datepicker** and it seems to be working fine

**Comment by smurfy** [comment]
@absalonv strange, last time i tried it failed with prod builds using aot and ivy. (not just angular9 you can disable ivy there)

I will test and report back.

**Comment by SchnWalter** [comment]
I'm closing this MR, the Angular Material v10 library provides a [date range selection](https://v10.material.angular.io/components/datepicker/overview#date-range-selection).

If someone needs any of these changes, feel free to use them. (The project is MIT licensed)

**Pull request closed without merging**